src/internal/v1api/modules.go

```diff
@@ -72,24 +73,8 @@ func (g Generator) readModuleMetadata(path string, logger *slog.Logger) (*module
 // writeModuleVersionListing writes the file containing the module version listing.
 // This data  is to be consumed when an end user requests /v1/modules/{namespace}/{name}/{targetSystem}/versions
 func (g Generator) writeModuleVersionListing(namespace string, name string, targetSystem string, versions []ModuleVersionResponseItem) error {
-	destinationDir := filepath.Join(g.DestinationDir, "v1", "modules", namespace, name, targetSystem)
-	if err := os.MkdirAll(destinationDir, 0755); err != nil {
-		return fmt.Errorf("failed to create directory: %w", err)
-	}
-
-	filePath := filepath.Join(destinationDir, "versions")
-
-	marshalled, err := json.Marshal(ModuleVersionListingResponse{Modules: []ModuleVersionListingResponseItem{{Versions: versions}}})
-	if err != nil {
-		return fmt.Errorf("failed to marshal json: %w", err)
-	}
-
-	err = os.WriteFile(filePath, marshalled, 0644)
-	if err != nil {
-		return fmt.Errorf("failed to write file: %w", err)
-	}
-
-	return nil
+	path := filepath.Join(g.DestinationDir, "v1", "modules", namespace, name, targetSystem, "versions")
+	return files.WriteToJsonFile(path, ModuleVersionListingResponse{Modules: []ModuleVersionListingResponseItem{{Versions: versions}}})
```

Comment:
It's a small nitpick on my side and it may seem like I'm being very hesitant unnecessarily, but I feel like "magic" functions that do more than what they're named are a bit of a code-smell and I would prefer to be explicit with the multiple stages here.

I'm happy to have a method to wrap making directories and writing to files, but I think using one single method to do both is a bit too "magic" for what is needed.

either that, or we need to rename the method to be something quite verbose to explain what it's doing.

As a developer approaching this code, I may question "Where is the directory made?" and not see this at a glance without having to dive in deep with how we're writing files to the disk. 

This is why I did not use this existing method in the files package. Infact I, personally, would like to remove the files package as I feel like it's not needed. 

However, I welcome others to propose alternative points of view

Reply:
I personally feel like this function is useful.

In practice the pattern of "serialize to a formatted json, make sure directory exists, write to file in that directory" will happen in quite a few places in the registry codebase.

No strong opinion about the packaging or function name, but the idea for the function itself, to me, is sound. And testing it makes sense to me as well (to make sure the function works when the directory doesn't exist, etc.).

I don't see value in repeating this code in every place it's needed.

Reply:
I'm happy to accept the value of this method, but then we should attempt to name it in a clear manner so that it's understandable to someone consuming this code what it does.

To keep it short im happy with:
- SafeDumpToJsonFile
- SafeWriteObjectToJsonFile

or something similar, this makes it a bit clearer that it's doing a little bit more than os.WriteFile under the hood.

Reply:
`SafeWriteObjectToJsonFile` works for me

Reply:
Changed in f69a096